bookkeeper-server/src/main/java/org/apache/bookkeeper/client/DefaultEnsemblePlacementPolicy.java

```diff
@@ -92,6 +94,9 @@
                     }
```

Comment:
is it possible for ```weightedSelection.getNextRandom()``` to return exactly the same one bookie which i.e. is already in the newBookies or excludeBookies thus resulting in the infinite loop?

Reply:
getNextRandom should eventually return different bookie. 

check the logic here - https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/WeightedRandomSelection.java#L149

Reply:
what if all the bookies already in either in excludeBookies or in a newBookies?

Reply:
there is check for this condition - https://github.com/apache/bookkeeper/pull/1788/files#diff-b8389e8cd72c81923a6c9aa82e119310L86